File: packages/cli/src/commands/udev.ts
import { confirm } from '../utils/prompts.js';
import { hasUdevRulesInstalled, installUdevRules } from '../utils/udev.js';
import { log } from '../utils/ui.js';

export async function udevCommand(options: { yes?: boolean }): Promise<void> {
  const installed = await hasUdevRulesInstalled();
  if (installed) {
    log.success('Wolf udev rules already installed');
    return;
  }

  const proceed = options.yes ? true : await confirm('Install Wolf udev rules?', true);
  if (!proceed) {
    log.warn('Skipped udev rules installation');
    return;
  }

  await installUdevRules();
  log.success('Wolf udev rules installed and reloaded');
}

File: packages/cli_native/src/commands/udev.rs
use anyhow::Result;

use crate::utils::{
    prompts::confirm,
    udev::{has_udev_rules_installed, install_udev_rules},
    ui::LOG,
};

pub async fn udev_command() -> Result<()> {
    let installed = has_udev_rules_installed().await;
    if installed {
        LOG.success("Wolf udev rules already installed");
        return Ok(());
    }

    let proceed = confirm("Install Wolf udev rules?", true)?;
    if !proceed {
        LOG.warn("Skipped udev rules installation");
        return Ok(());
    }

    install_udev_rules().await?;
    LOG.success("Wolf udev rules installed and reloaded");
    Ok(())
}
